components/gantry/multiaxis/multiaxis.go

```diff
@@ -115,18 +116,28 @@ func (g *multiAxis) MoveToPosition(
 		return errors.Errorf("need position inputs for %v-axis gantry, have %v positions", len(g.subAxes), len(positions))
 	}
 
-	idx := 0
+	if len(positions) != len(g.lengthsMm) {
+		return errors.Errorf(
+			"number of input positions %v does not match total gantry axes length %v",
```

Comment:
Change "length" to "count." It's too easy to misread this and think that the position is supposed to be the farthest position out on the gantry (its length). (Other rephrasings are possible, too. Go with whatever you think makes it clearest that the problem is that the input has the wrong dimensionality.)